custom_components/elmo_iess_alarm/binary_sensor.py

```diff
@@ -52,6 +52,7 @@ def __init__(
     ) -> None:
         """Construct."""
         super().__init__(coordinator)
+        self._entity_id = f"{DOMAIN}.{name}"
```

Comment:
This line actually changes all entities IDs.

Side note: this assignment should return a name like `binary_sensor.elmo_iess_alarm.1_tamper_sirena`, but the ID is automatically translated into `binary_sensor.elmo_iess_alarm_1_tamper_sirena` as dots (`.`) are not considered valid (HA should have a renaming convention).

I would suggest to keep the dot as it is also recommended in their FAQs: https://www.home-assistant.io/faq/unique_id/ (see entity_id section where they recommend `<domain>.<id>`).